### ISSUE TITLE
Outcome incorrectly set by Flank

### ISSUE THREAD

**Issue by bootstraponline** [Flank/flank#914]
**Describe the bug**
> A clear and concise description of what the bug is.

```
[task 2020-07-23T20:29:15.152Z] ┌─────────┬──────────────────────┬──────────────────────────────────────────┐
[task 2020-07-23T20:29:15.153Z] │ OUTCOME │      MATRIX ID       │               TEST DETAILS               │
[task 2020-07-23T20:29:15.153Z] ├─────────┼──────────────────────┼──────────────────────────────────────────┤
[task 2020-07-23T20:29:15.153Z] │  flaky  │ matrix-1vj2pt9wih182 │ 1 test cases failed, 106 passed, 2 flaky │
[task 2020-07-23T20:29:15.153Z] └─────────┴──────────────────────┴──────────────────────────────────────────┘
```

FTL confirmed the matrix outcome is actually failure. Flank reported flaky.

Let's diagram the Flank outcome setting logic with PlantUML in addition to fixing this bug.

**Expected behavior**
> A clear and concise description of what you expected to happen.

I expected the outcome to match the FTL outcome (failure).

**Details (please complete the following information):**
> Have you tested on the latest Flank snapshot?
>
> Post the output of ` flank --version`.

Flank  v20.07.0

**Additional context**
> Add any other context about the problem here.

https://firebase-community.slack.com/archives/C72V6UW8M/p1595539705001900

**Comment by pawelpasterz** [comment]
The logic that flank needs to reflect:

- at least one test failed -> `failed`
- no failed test but at least 1 flaky -> `flaky` **BUT** FTL converts it to `success`
- no failed and no flaky -> `success`

**Comment by pawelpasterz** [comment]
#### `gcloud` output
##### (1) at least one test failed
![Screenshot 2020-07-28 at 09 01 41](https://user-images.githubusercontent.com/32893017/88630693-4abe5d80-d0b1-11ea-8ac1-c6f937f8d936.png)
##### (2) no failed test but at least 1 flaky
![Screenshot 2020-07-28 at 09 01 56](https://user-images.githubusercontent.com/32893017/88630729-56aa1f80-d0b1-11ea-99c7-91c0c2ae91f2.png)
##### (3) no failed and no flaky
![Screenshot 2020-07-28 at 09 02 09](https://user-images.githubusercontent.com/32893017/88630743-5dd12d80-d0b1-11ea-9ce8-18e8eaf45fcf.png)

**Comment by pawelpasterz** [comment]
#### `flank` output
##### at least one test failed
![Screenshot 2020-07-28 at 09 33 40](https://user-images.githubusercontent.com/32893017/88633628-8e1acb00-d0b5-11ea-84b0-1071f1b24365.png)
![Screenshot 2020-07-28 at 09 18 09](https://user-images.githubusercontent.com/32893017/88633658-95da6f80-d0b5-11ea-80e2-37cd4a162639.png)
###### no failed test but at least one flaky
![Screenshot 2020-07-28 at 09 13 15](https://user-images.githubusercontent.com/32893017/88633813-c7ebd180-d0b5-11ea-8924-83dcb082ef4e.png)
##### no failed and no flaky
![Screenshot 2020-07-28 at 09 23 23](https://user-images.githubusercontent.com/32893017/88633843-d33efd00-d0b5-11ea-93c4-abcb4a9f62e5.png)


What needs to be fixed:
- the failed test always means failed outcome, regardless of others being flaky (currently flaky tests obscure failed ones)
- flaky tests mean flaky outcome but with success exit code (currently exit code is `0` but status is `success`)
- optional: do we want to change the naming of the outcome? (ex. `failure` -> `failed`) cc @jan-gogo

**Comment by bootstraponline** [comment]
> * do we want to change the naming of the outcome?

for naming, let's match gcloud.

**Comment by pawelpasterz** [comment]
> no failed test but at least one flaky EDIT: below outcome is Corner Case
![Screenshot 2020-07-28 at 09 13 15](https://user-images.githubusercontent.com/32893017/88633813-c7ebd180-d0b5-11ea-8924-83dcb082ef4e.png)

Above is a corner case that needs to be resolved in a separate issue. It requires to check flank API calls with gcloud's.
Originally described bug to be resolved in #917

**Comment by pawelpasterz** [comment]
@rpappalax 
Fix in review #917. Please let me know immediately if you will encounter a similar issue in the future. Please, keep in mind there is another issue found (thanks to your case 😄  ) we still need to investigate #918
